libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -1398,6 +1404,41 @@ def invoke(self, input: Input, config: Optional[RunnableConfig] = None) -> Outpu
         return self._call_with_config(self.func, input, config)
 
 
+class RunnableEach(Serializable, Runnable[List[Input], List[Output]]):
+    """
+    A runnable that delegates calls to another runnable with each element of the input sequence.
+    """
+
+    bound: Runnable[Input, Output]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @property
+    def lc_serializable(self) -> bool:
+        return True
+
+    @property
+    def lc_namespace(self) -> List[str]:
+        return self.__class__.__module__.split(".")[:-1]
+
+    def each(self) -> RunnableEach[Input, Output]:  # type: ignore[override]
```

Comment:
The problem is Mypy wants this to be `Runnable[List[List[Input]], List[List[Output]]]`

Reply:
Unfortunately I think mypy is correct here:
- `Runnable.map` has a signature of `-> Runnable[List[Input], List[Output]]`.
- `RunnableEach[I, O]` is a `Runnable[List[I], List[O]]`.
- That means that in order to subtype `Runnable`, `RunnableEach[I, O].map` has to be `-> Runnable[List[List[I]], List[List[O]]]`.

I think this might cause further issues in downstream code, both inside langchain and in downstream packages that depend on it.

It feels like `RunnableEach` shouldn't be `Runnable`, and perhaps there's a missing abstraction that should sit underneath both of them?